src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -524,24 +524,28 @@ private enum NamespaceParts
                         case SyntaxKind.OpenBracketToken:
                             if (this.IsPossibleGlobalAttributeDeclaration())
                             {
-                                // incomplete members must be processed before we add any nodes to the body:
-                                ReduceIncompleteMembers(ref pendingIncompleteMembers, ref openBraceOrSemicolon, ref body, ref initialBadNodes);
-
-                                var attribute = this.ParseAttributeDeclaration();
-                                if (!isGlobal || seen > NamespaceParts.GlobalAttributes)
-                                {
-                                    RoslynDebug.Assert(attribute.Target != null, "Must have a target as IsPossibleGlobalAttributeDeclaration checks for that");
-                                    attribute = this.AddError(attribute, attribute.Target.Identifier, ErrorCode.ERR_GlobalAttributesNotFirst);
-                                    this.AddSkippedNamespaceText(ref openBraceOrSemicolon, ref body, ref initialBadNodes, attribute);
-                                }
-                                else
+                                // Could be an attribute, or it could be a collection literal at the top level.  e.g. `[1, 2, 3].XYZ();`
```

Comment:
>or it could be a collection literal at the top level.  e.g. `[1, 2, 3].XYZ();` [](http://example.com/codeflow?start=58&length=78)

It looks like `IsPossibleGlobalAttributeDeclaration()` only returns true for `[assembly:` and `[module:`. Is it possible to get here with a collection literal?

Reply:
> It looks like IsPossibleGlobalAttributeDeclaration() only returns true for [assembly: or [module:. Is it possible to get here with a collection literal?

Yes, and these would be good test cases @jnm2 .  Specifically, this is a legal dictionary literal:

```c#
int assembly = 0;
[assembly: 1].DoWhatever();
```

Reply:
Thanks. Consider updating the example in the comment: perhaps `e.g. [assembly: X].Y`

Reply:
Done.  and added tests.